File: src/core/main/java/workschedule/Chromosome.java
package workschedule;

public final class Chromosome {
    private final boolean[] params;

    public Chromosome(final int length) {
        params = new boolean[length];
    }

    public boolean getParam(final int index) {
        return params[index];
    }

    public void setParam(final int index, final boolean value) {
        params[index] = value;
    }
}

File: src/core/legacy/ga/Chromosome.js

'use strict';

class Chromosome {

  constructor(length) {
    this.length = length;
    this.eachParam(i => this[i] = false);
  }

  eachParam(cb) {
    for (let i = 0; i < this.length; i += 1) {
      cb.call(this, i, this.getParam(i));
    }

    return this;
  }

  fitness() {
    return 0;
  }

  setParam(i, value) {
    this[i] = value;
    return this;
  }

  getParam(i) {
    return this[i];
  }

}

module.exports = Chromosome;
